src/Run/Setup.cs

```diff
@@ -14,13 +18,51 @@ public class Setup
         public string Os { get; set; }
         public string ConfigurationFilePath { get; set; }
 
+        private string ParseSettingValue(string inputValue)
+        {
+            string value = string.Empty;
+            int length = inputValue.Length;
+            for (int i = 0; i < length; i++)
+            {
+                if (inputValue[i] == '$')
+                {
+                    if (inputValue[i + 1] == '{')
```

Comment:
This will crash for any strings that end in `$` we need to check to be sure `i+1` isn't off the end of the string.